File: eureka/android/library/src/main/java/com/lifeplus/Pojo/UserSessionStruct.java
package com.lifeplus.Pojo;

import java.util.Date;

public class UserSessionStruct {
    private final String id;
    private final String user_id;
    private final String device_id;
    private final Date login_date;
    private final Date logout_date;
    private final String auth_token;
    private final String refresh_token;
    private final String gateway_token;

    public UserSessionStruct(String pId, String pUserId, String pDeviceId, Date pLoginDate, Date pLogoutDate,
                             String pAuthToken, String pRefreshToken, String pGatewayToken) {
        id = pId;
        user_id = pUserId;
        device_id = pDeviceId;
        login_date = pLoginDate;
        logout_date = pLogoutDate;
        auth_token = pAuthToken;
        refresh_token = pRefreshToken;
        gateway_token = pGatewayToken;
    }

    public String getSessionId() {
        return id;
    }

    public String getUserId() {
        return user_id;
    }

    public String getDeviceId() {
        return device_id;
    }

    public Date getLoginDate() {
        return login_date;
    }

    public Date getLogoutDate() {
        return logout_date;
    }

    public String getAuthToken() {
        return auth_token;
    }

    public String getRefreshToken() {
        return refresh_token;
    }

    public String getGatewayToken() {
        return gateway_token;
    }
}

File: eureka/ios/nativeLib/nativeLib/Models/UserSessionStruct.swift
//
//  UserSessionStruct.swift
//  BLEDemo
//

import Foundation

public class UserSessionStruct {
    
  private final var id: String
  private final var user_id: String
  private final var device_id: String
  private final var login_date: Date?
  private final var logout_date: Date?
  private final var auth_token: String
  private final var refresh_token: String
  private final var gateway_token: String
    
  internal init(pId: String, pUserId: String, pDeviceId: String, pLoginDate: Date?, pLogoutDate: Date?,
                pAuthToken: String, pRefreshToken: String, pGatewayToken: String) {
    id = pId;
    user_id = pUserId;
    device_id = pDeviceId;
    login_date = pLoginDate;
    logout_date = pLogoutDate;
    auth_token = pAuthToken;
    refresh_token = pRefreshToken;
    gateway_token = pGatewayToken;
  }
    
  internal func getSessionId() -> String {
    return id
  }
    
  internal func getUserId() -> String {
    return user_id
  }
    
  internal func getDeviceId() -> String {
    return device_id
  }
    
  internal func getLoginDate() -> Date? {
    return login_date
  }
    
  internal func getLogoutDate() -> Date? {
    return logout_date
  }
    
  internal func getAuthToken() -> String {
    return auth_token;
  }
    
  internal func getRefreshToken() -> String {
    return refresh_token
  }
    
  internal func getGatewayToken() -> String {
    return gateway_token
  }
}
